### PR TITLE
Queries serialization improvements

### PR DESCRIPTION
**Pull request by irakliyk**
This PR addresses #6 and also improves queries serialization/deserialization overall. Specifically:

- [x] Implement deserialization of query bytes directly into field elements
- [x] Implement flat serialization of query values and Merkle paths (for trace and constraint queries only)

The overall impact of this PR is ~1% reduction in proof sizes, but it also lays groundwork for switching over to new Merkle tree implementation.

### CODE REVIEW

File: common/src/proof.rs [hunk 1]
```diff
@@ -113,34 +117,77 @@ impl StarkProof {
...
 
 impl Queries {
-    /// Returns a set of queries constructed from a batch Merkle proof and corresponding values.
-    pub fn new<E: FieldElement>(merkle_proof: BatchMerkleProof, values: Vec<Vec<E>>) -> Self {
+    /// Returns a set of queries constructed from a batch Merkle proof and corresponding elements.
+    pub fn new<E: FieldElement>(merkle_proof: BatchMerkleProof, query_values: Vec<Vec<E>>) -> Self {
+        assert!(!query_values.is_empty(), "query values cannot be empty");
+        let elements_per_query = query_values[0].len();
+        assert!(
+            elements_per_query != 0,
```
Comment (kchalkias) [inline common/src/proof.rs:125]: prefer `is_empty`

File: common/src/proof.rs [hunk 2]
```diff
@@ -113,34 +117,77 @@ impl StarkProof {
...
+        let paths = merkle_proof.serialize_nodes();
+
+        Queries { paths, values }
     }
 
-    /// Convert a set of queries into a batch Merkle proof and corresponding values.
-    /// TODO: return values as a vector of field elements
-    pub fn into_batch<H: Hasher>(self, num_leaves: usize) -> (BatchMerkleProof, Vec<Vec<u8>>) {
+    /// Convert a set of queries into a batch Merkle proof and corresponding query values.
+    pub fn deserialize<H: Hasher, E: FieldElement>(
```
Comment (kchalkias) [inline common/src/proof.rs:151]: Can any zero value or length cause issues or panic (ie division by zero?)

**Review by kchalkias — APPROVED** [review]
minor comments

File: common/src/proof.rs (hunk 2 above)
Comment (irakliyk) [reply common/src/proof.rs:151]: Good point! I've added some assertions here as this code gets called internally from within Winterfell. So, if it is called with wrong parameters, it's ok to panic.

File: common/src/proof.rs (hunk 1 above)
Comment (irakliyk) [reply common/src/proof.rs:125]: I usually do too - but in this case we need to cache the `elements_per_query` value - so, I think it's OK to have it like this.

**Pull request merged**